sylt-compiler/src/typechecker.rs

```diff
@@ -1357,6 +1454,35 @@ impl TypeChecker {
                     }
                 }
 
+                (Type::Enum(a_name, a_variants), Type::Blob(b_name, b_variants)) => {
```

Comment:
```suggestion
                (Type::Enum(a_name, a_variants), Type::Enum(b_name, b_variants)) => {
```
They're that similar huh?

Reply:
owo

Reply:
They're similar - but they're semantics are completely different. Integers and pointers have the same representation in memory - should they also be equal?